components/collator/src/comparison.rs

```diff
@@ -262,8 +183,152 @@ impl Collator {
         let mut merged_options = CollatorOptionsBitField::from(options);
         merged_options.set_defaults(altered_defaults);
 
-        let special_primaries = if merged_options.alternate_handling() == AlternateHandling::Shifted
-            || merged_options.numeric()
+        Ok(LocaleSpecificDataHolder {
+            tailoring,
+            diacritics,
+            merged_options,
+            reordering,
+            lithuanian_dot_above: metadata.lithuanian_dot_above(),
+        })
+    }
+}
+
+/// Compares strings according to culturally-relevant ordering.
+#[derive(Debug)]
+pub struct Collator {
+    special_primaries: Option<DataPayload<CollationSpecialPrimariesV1Marker>>,
+    root: DataPayload<CollationRootV1Marker>,
+    tailoring: Option<DataPayload<CollationTailoringV1Marker>>,
+    jamo: DataPayload<CollationJamoV1Marker>,
+    diacritics: DataPayload<CollationDiacriticsV1Marker>,
+    options: CollatorOptionsBitField,
+    reordering: Option<DataPayload<CollationReorderingV1Marker>>,
+    decompositions: DataPayload<CanonicalDecompositionDataV1Marker>,
+    tables: DataPayload<CanonicalDecompositionTablesV1Marker>,
+    lithuanian_dot_above: bool,
+}
+
+impl Collator {
+    /// Constructs a borrowed version of this type for more efficient querying.
+    pub fn as_borrowed(&self) -> CollatorBorrowed {
+        CollatorBorrowed {
+            special_primaries: self.special_primaries.as_ref().map(|s| s.get()),
+            root: self.root.get(),
+            tailoring: self.tailoring.as_ref().map(|s| s.get()),
+            jamo: self.jamo.get(),
+            diacritics: self.diacritics.get(),
+            options: self.options,
+            reordering: self.reordering.as_ref().map(|s| s.get()),
+            decompositions: self.decompositions.get(),
+            tables: self.tables.get(),
+            lithuanian_dot_above: self.lithuanian_dot_above,
+        }
+    }
+
+    /// Creates a collator for the given locale and options from compiled data.
+    ///
+    /// Unless you know you need this constructor, it is likely that `CollatorBorrowed::try_new`
+    /// is more appropriate.
+    #[cfg(feature = "compiled_data")]
+    pub fn try_new(locale: &DataLocale, options: CollatorOptions) -> Result<Self, DataError> {
```

Comment:
Won't that break FFI?

Reply:
We have a `static_to_owned` method on other borrowed types that wraps in a `DataPayload`.

Reply:
This is how we do it for other types: https://github.com/unicode-org/icu4x/blob/b7d567f2dc84cac30397a2d24d7e2390bc9caff3/ffi/capi/src/fallbacker.rs#L63

Reply:
I made this change, but I commented out the tests that tested owned instantiation, since I failed to figure out how to satisfy the trait bounds of `Collator::try_new_unstable` with baked data.

Reply:
Marking this resolved, since the review comment itself is resolved even if a new issue (using `try_new_unstable` in owned-construction tests) showed up.